.github/workflows/android.yml

```diff
@@ -35,11 +35,308 @@ jobs:
           sudo apt-get -qq update
           sudo apt-get install -y libncurses5
 
-      - name: Setup Java
-        uses: actions/setup-java@v3
+      - name: Setup Android SDK
+        uses: android-actions/setup-android@v2
+
+      - name: Cache dependencies
+        uses: actions/cache@v3
         with:
-          distribution: 'zulu'
-          java-version: "17"
+          path: |
+            3rdParty/buildCache
+            !3rdParty/buildCache/android/vcpkgcache/
+          key: android-${{ matrix.arch }}-${{ hashFiles('android/*.sh', '.github/workflows/android.yml') }}
+
+      - name: Unset ANDROID_NDK_ROOT environment variable
+        if: ${{ success() }}
+        run: |
+          echo ANDROID_NDK_ROOT="" >> $GITHUB_ENV
+
+      - name: Build dependencies
+        if: ${{ success() }}
+        run:  |
+          ./android/ci_build_dependencies.sh --arch ${{ matrix.arch }}
+
+      - name: Prepare logs on failure
+        if: ${{ failure() }}
+        run: python ./deploy/prepare_deployment.py logs
+
+      - name: Upload logs on failure
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: android_logs_dependencies_${{ matrix.arch }}_${{ github.event.pull_request.head.sha }}
+          path: deploy/logs.7z
+
+  build-dependencies-vcpkg:
```

Comment:
In case in vcpkg, don't use dependency. If the ports not install, it will make them compile twice.

Reply:
Compile twice
![image](https://user-images.githubusercontent.com/7043539/225159745-d2b481a8-33da-4ba5-bb60-ad3499703efa.png)